safe_transaction_service/history/services/collectibles_service.py

```diff
@@ -161,7 +161,41 @@ def __init__(self, ethereum_client: EthereumClient, redis: Redis):
         self.ethereum_client = ethereum_client
         self.ethereum_network = ethereum_client.get_network()
         self.redis = redis
-        self.ens_service: EnsClient = EnsClient(self.ethereum_network.value)
+
+        base_url = settings.ENS_SUBGRAPH_URL
+        api_key = settings.ENS_SUBGRAPH_API_KEY
+        subgraph_id = settings.ENS_SUBGRAPH_ID
+
+        if self.ethereum_network == EthereumNetwork.SEPOLIA:
+            config = EnsClient.Config(
+                EthereumNetwork.SEPOLIA,
+                "https://api.studio.thegraph.com/query/49574/enssepolia/version/latest",
+            )
+        elif self.ethereum_network == EthereumNetwork.HOLESKY:
+            config = EnsClient.Config(
+                EthereumNetwork.HOLESKY,
+                "https://api.studio.thegraph.com/query/49574/ensholesky/version/latest",
+            )
+        elif base_url and api_key and subgraph_id:
+            config = EnsClient.SubgraphConfig(
+                network=EthereumNetwork.MAINNET,
+                base_url=base_url,
+                api_key=api_key,
+                subgraph_id=subgraph_id,
+            )
+        else:
```

Comment:
I don't think that we need this sequence of ifs.
```suggestion
        if  base_url: 
            config = EnsClient.Config(self.ethereum_network, base_ur, api_key, subgraph_idl)
        else: 
            logger.warning()
            # Default fallback
```

Reply:
Not sure if I follow. If we only check that the `base_url` is not `None` then `api_key` and `subgraph_id` might be `None` and we should not use the `EnsClient.SubgraphConfig` 🤔

Reply:
Understood but would you find other way to avoid chose between networks in the endpoint?

Reply:
Restructured the control flow in [41d316c](https://github.com/safe-global/safe-transaction-service/pull/2105/commits/41d316c640eb4fdb29d83c8a1f67a142d6930cde)

Given that the chain id is no longer required I think the current state is now clearer.